tests/unit/test_container_decoders.py

```diff
@@ -42,3 +44,54 @@ def test_project_with_datasets_decoder(self, project_with_datasets):
         assert dataset_2.id.val == 2L
         assert dataset_2.name.val == 'dataset_name_2'
         assert dataset_2.description.val == 'dataset_description_2'
+
+
+class TestScreenDecoder(object):
+
+    def assert_screen(self, screen):
+        assert screen.name.val == 'the_name'
+        assert screen.description.val == 'the_description'
+        assert screen.protocolDescription.val == 'the_protocol_description'
+        assert screen.protocolIdentifier.val == 'the_protocol_identifier'
+        assert screen.reagentSetDescription.val == 'the_reagent_set_description'
+        assert screen.reagentSetDescription.val == 'the_reagent_set_description'
+        assert screen.reagentSetIdentifier.val == 'the_reagent_set_identifier'
+        assert screen.type.val == 'the_type'
+
+    def assert_plate(self, plate, plate_id):
+        assert plate.id.val == plate_id
+        assert plate.name.val == 'plate_name_%d' % plate_id
+        assert plate.description.val == 'plate_description_%d' % plate_id
+        assert plate.columnNamingConvention.val == 'number'
+        assert plate.rowNamingConvention.val == 'letter'
+        assert plate.columns.val == 12
+        assert plate.rows.val == 8
+        assert plate.defaultSample.val == 0
+        assert plate.externalIdentifier.val == 'external_identifier_%d' % plate_id
+        assert plate.status.val == 'status_%d' % plate_id
+        assert plate.wellOriginX.__class__ is LengthI
+        assert plate.wellOriginX.getUnit() == UnitsLength.REFERENCEFRAME
+        assert plate.wellOriginX.getValue() == 0.1
+        assert plate.wellOriginY.__class__ is LengthI
+        assert plate.wellOriginY.getUnit() == UnitsLength.REFERENCEFRAME
+        assert plate.wellOriginY.getValue() == 1.1
+
+    def test_screen_decoder(self, screen):
+        encoder = get_encoder(screen.__class__)
+        decoder = get_decoder(encoder.TYPE)
+        v = encoder.encode(screen)
+        v = decoder.decode(v)
+        self.assert_screen(v)
+        assert v.sizeOfPlateLinks() == 0
+
+    def test_screen_with_plates_decoder(self, screen_with_plates):
+        encoder = get_encoder(screen_with_plates.__class__)
+        decoder = get_decoder(encoder.TYPE)
+        v = encoder.encode(screen_with_plates)
+        v = decoder.decode(v)
+        self.assert_screen(v)
+        assert v.id.val == 4L
+        assert v.sizeOfPlateLinks() == 2
+        plate_1, plate_2 = v.linkedPlateList()
+        self.assert_plate(plate_1, 5)
```

Comment:
Probably being pedantic as it won't affect the assertion but for clarity these should be `5L` and `6L`.